alica_engine/include/engine/BasicBehaviour.h

```diff
@@ -73,6 +75,8 @@ class BasicBehaviour
 
     bool isEventDriven() const { return _behaviour->isEventDriven(); }
 
+    amr_interfaces::TraceContext getTraceContext() { return _trace->context(); };
```

Comment:
can instead return a reference to the trace object itself i.e. IAlicaTrace&. That way the application can set logs & tags into the trace while the behaviour/plan is active.